description: disable tracing, it depends on too many sourceless javascript files
author: Michael Gilbert <mgilbert@debian.org>

--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1794,18 +1794,6 @@ jumbo_split_static_library("browser") {
     "task_manager/web_contents_tags.h",
     "themes/theme_service_win.cc",
     "themes/theme_service_win.h",
-    "tracing/background_tracing_field_trial.cc",
-    "tracing/background_tracing_field_trial.h",
-    "tracing/background_tracing_metrics_provider.cc",
-    "tracing/background_tracing_metrics_provider.h",
-    "tracing/chrome_tracing_delegate.cc",
-    "tracing/chrome_tracing_delegate.h",
-    "tracing/crash_service_uploader.cc",
-    "tracing/crash_service_uploader.h",
-    "tracing/navigation_tracing.cc",
-    "tracing/navigation_tracing.h",
-    "tracing/trace_event_system_stats_monitor.cc",
-    "tracing/trace_event_system_stats_monitor.h",
     "transition_manager/full_browser_transition_manager.cc",
     "transition_manager/full_browser_transition_manager.h",
     "translate/android/translate_utils.cc",
@@ -2100,7 +2088,6 @@ jumbo_split_static_library("browser") {
     "//components/sync_preferences",
     "//components/sync_sessions",
     "//components/tab_count_metrics",
-    "//components/tracing:startup_tracing",
     "//components/translate/content/browser",
     "//components/translate/core/browser",
     "//components/translate/core/common",
--- a/content/browser/BUILD.gn
+++ b/content/browser/BUILD.gn
@@ -67,8 +67,6 @@ jumbo_source_set("browser") {
     "//components/services/filesystem:lib",
     "//components/services/leveldb:lib",
     "//components/services/quarantine:quarantine",
-    "//components/tracing",
-    "//components/tracing:startup_tracing",
     "//components/tracing/common:interfaces",
     "//components/ui_devtools",
     "//components/url_formatter",
@@ -177,9 +175,6 @@ jumbo_source_set("browser") {
     "//services/service_manager/zygote:zygote_buildflags",
     "//services/shape_detection:lib",
     "//services/shape_detection/public/mojom",
-    "//services/tracing:lib",
-    "//services/tracing:manifest",
-    "//services/tracing/public/cpp",
     "//services/video_capture:lib",
     "//services/video_capture/public/cpp",
     "//services/video_capture/public/mojom:constants",
@@ -764,8 +759,6 @@ jumbo_source_set("browser") {
     "devtools/protocol/target_handler.h",
     "devtools/protocol/tethering_handler.cc",
     "devtools/protocol/tethering_handler.h",
-    "devtools/protocol/tracing_handler.cc",
-    "devtools/protocol/tracing_handler.h",
     "devtools/render_frame_devtools_agent_host.cc",
     "devtools/render_frame_devtools_agent_host.h",
     "devtools/service_worker_devtools_agent_host.cc",
@@ -1798,29 +1791,6 @@ jumbo_source_set("browser") {
     "theme_helper.h",
     "theme_helper_mac.h",
     "theme_helper_mac.mm",
-    "tracing/background_memory_tracing_observer.cc",
-    "tracing/background_memory_tracing_observer.h",
-    "tracing/background_startup_tracing_observer.cc",
-    "tracing/background_startup_tracing_observer.h",
-    "tracing/background_tracing_active_scenario.cc",
-    "tracing/background_tracing_active_scenario.h",
-    "tracing/background_tracing_agent_client_impl.cc",
-    "tracing/background_tracing_agent_client_impl.h",
-    "tracing/background_tracing_config_impl.cc",
-    "tracing/background_tracing_config_impl.h",
-    "tracing/background_tracing_manager_impl.cc",
-    "tracing/background_tracing_manager_impl.h",
-    "tracing/background_tracing_rule.cc",
-    "tracing/background_tracing_rule.h",
-    "tracing/file_tracing_provider_impl.cc",
-    "tracing/file_tracing_provider_impl.h",
-    "tracing/perfetto_file_tracer.cc",
-    "tracing/perfetto_file_tracer.h",
-    "tracing/tracing_controller_impl.cc",
-    "tracing/tracing_controller_impl.h",
-    "tracing/tracing_controller_impl_data_endpoint.cc",
-    "tracing/tracing_ui.cc",
-    "tracing/tracing_ui.h",
     "url_loader_factory_getter.cc",
     "url_loader_factory_getter.h",
     "utility_process_host.cc",
@@ -2060,7 +2030,6 @@ jumbo_source_set("browser") {
   if (!is_android) {
     deps += [
       "//components/vector_icons",
-      "//content/browser/tracing:resources",
     ]
   }
 
--- a/content/shell/BUILD.gn
+++ b/content/shell/BUILD.gn
@@ -513,7 +513,6 @@ repack("pak") {
   sources = [
     "$root_gen_dir/content/app/resources/content_resources_100_percent.pak",
     "$root_gen_dir/content/app/strings/content_strings_en-US.pak",
-    "$root_gen_dir/content/browser/tracing/tracing_resources.pak",
     "$root_gen_dir/content/content_resources.pak",
     "$root_gen_dir/content/shell/shell_resources.pak",
     "$root_gen_dir/mojo/public/js/mojo_bindings_resources.pak",
@@ -531,7 +530,6 @@ repack("pak") {
     "//content:resources",
     "//content/app/resources",
     "//content/app/strings",
-    "//content/browser/tracing:resources",
     "//mojo/public/js:resources",
     "//net:net_resources",
     "//third_party/blink/public:resources",
--- a/headless/BUILD.gn
+++ b/headless/BUILD.gn
@@ -35,7 +35,6 @@ repack("pak") {
     "$root_gen_dir/content/app/resources/content_resources_100_percent.pak",
     "$root_gen_dir/content/app/strings/content_strings_en-US.pak",
     "$root_gen_dir/content/browser/devtools/devtools_resources.pak",
-    "$root_gen_dir/content/browser/tracing/tracing_resources.pak",
     "$root_gen_dir/content/content_resources.pak",
     "$root_gen_dir/headless/headless_lib_resources.pak",
     "$root_gen_dir/mojo/public/js/mojo_bindings_resources.pak",
@@ -62,7 +61,6 @@ repack("pak") {
     "//content/app/resources",
     "//content/app/strings",
     "//content/browser/devtools:resources",
-    "//content/browser/tracing:resources",
     "//mojo/public/js:resources",
     "//net:net_resources",
     "//third_party/blink/public:resources",
--- a/chrome/chrome_paks.gni
+++ b/chrome/chrome_paks.gni
@@ -99,7 +99,6 @@ template("chrome_extra_paks") {
       "$root_gen_dir/chrome/quota_internals_resources.pak",
       "$root_gen_dir/components/autofill/core/browser/autofill_address_rewriter_resources.pak",
       "$root_gen_dir/components/components_resources.pak",
-      "$root_gen_dir/content/browser/tracing/tracing_resources.pak",
       "$root_gen_dir/content/content_resources.pak",
       "$root_gen_dir/mojo/public/js/mojo_bindings_resources.pak",
       "$root_gen_dir/net/net_resources.pak",
@@ -116,7 +115,6 @@ template("chrome_extra_paks") {
       "//components/autofill/core/browser:autofill_address_rewriter_resources",
       "//components/resources",
       "//content:resources",
-      "//content/browser/tracing:resources",
       "//mojo/public/js:resources",
       "//net:net_resources",
       "//third_party/blink/public:resources",
--- a/content/public/browser/background_tracing_config.cc
+++ b/content/public/browser/background_tracing_config.cc
@@ -4,8 +4,6 @@
 
 #include "content/public/browser/background_tracing_config.h"
 
-#include "content/browser/tracing/background_tracing_config_impl.h"
-
 namespace content {
 
 BackgroundTracingConfig::BackgroundTracingConfig(TracingMode tracing_mode)
@@ -15,7 +13,7 @@ BackgroundTracingConfig::~BackgroundTrac
 
 std::unique_ptr<BackgroundTracingConfig> BackgroundTracingConfig::FromDict(
     const base::DictionaryValue* dict) {
-  return BackgroundTracingConfigImpl::FromDict(dict);
+  return nullptr;
 }
 
 }  // namespace content
--- a/content/public/browser/background_tracing_config.h
+++ b/content/public/browser/background_tracing_config.h
@@ -37,7 +37,6 @@ class CONTENT_EXPORT BackgroundTracingCo
   virtual void IntoDict(base::DictionaryValue* dict) = 0;
 
  private:
-  friend class BackgroundTracingConfigImpl;
   explicit BackgroundTracingConfig(TracingMode tracing_mode);
 
   const TracingMode tracing_mode_;
--- a/content/browser/browser_main_loop.cc
+++ b/content/browser/browser_main_loop.cc
@@ -89,8 +89,6 @@
 #include "content/browser/speech/speech_recognition_manager_impl.h"
 #include "content/browser/startup_data_impl.h"
 #include "content/browser/startup_task_runner.h"
-#include "content/browser/tracing/background_tracing_manager_impl.h"
-#include "content/browser/tracing/tracing_controller_impl.h"
 #include "content/browser/utility_process_host.h"
 #include "content/browser/webrtc/webrtc_internals.h"
 #include "content/browser/webui/content_web_ui_controller_factory.h"
@@ -972,10 +970,6 @@ int BrowserMainLoop::CreateThreads() {
 }
 
 int BrowserMainLoop::PostCreateThreads() {
-  tracing_controller_ = std::make_unique<content::TracingControllerImpl>();
-  content::BackgroundTracingManagerImpl::GetInstance()
-      ->AddMetadataGeneratorFunction();
-
   if (parts_) {
     TRACE_EVENT0("startup", "BrowserMainLoop::PostCreateThreads");
     parts_->PostCreateThreads();
@@ -1522,13 +1516,6 @@ void BrowserMainLoop::InitializeMojo() {
   // know they're running in the same process as the service.
   content::NavigableContentsView::SetClientRunningInServiceProcess();
 
-  // Start startup tracing through TracingController's interface. TraceLog has
-  // been enabled in content_main_runner where threads are not available. Now We
-  // need to start tracing for all other tracing agents, which require threads.
-  // We can only do this after starting the main message loop to avoid calling
-  // MessagePumpForUI::ScheduleWork() before MessagePumpForUI::Start().
-  TracingControllerImpl::GetInstance()->StartStartupTracingIfNeeded();
-
 #if BUILDFLAG(MOJO_RANDOM_DELAYS_ENABLED)
   mojo::BeginRandomMojoDelays();
 #endif
--- a/content/browser/browser_main_loop.h
+++ b/content/browser/browser_main_loop.h
@@ -390,7 +390,6 @@ class CONTENT_EXPORT BrowserMainLoop {
 
   std::unique_ptr<MediaStreamManager> media_stream_manager_;
   scoped_refptr<SaveFileManager> save_file_manager_;
-  std::unique_ptr<content::TracingControllerImpl> tracing_controller_;
   scoped_refptr<responsiveness::Watcher> responsiveness_watcher_;
 #if !defined(OS_ANDROID)
   // A SharedBitmapManager used to sharing and mapping IDs to shared memory
--- a/content/browser/devtools/browser_devtools_agent_host.cc
+++ b/content/browser/devtools/browser_devtools_agent_host.cc
@@ -20,7 +20,6 @@
 #include "content/browser/devtools/protocol/system_info_handler.h"
 #include "content/browser/devtools/protocol/target_handler.h"
 #include "content/browser/devtools/protocol/tethering_handler.h"
-#include "content/browser/devtools/protocol/tracing_handler.h"
 #include "content/browser/frame_host/frame_tree_node.h"
 
 namespace content {
@@ -89,8 +88,6 @@ bool BrowserDevToolsAgentHost::AttachSes
     session->AddHandler(std::make_unique<protocol::TetheringHandler>(
         socket_callback_, tethering_task_runner_));
   }
-  session->AddHandler(
-      std::make_unique<protocol::TracingHandler>(nullptr, GetIOContext()));
   return true;
 }
 
--- a/content/browser/devtools/render_frame_devtools_agent_host.cc
+++ b/content/browser/devtools/render_frame_devtools_agent_host.cc
@@ -40,7 +40,6 @@
 #include "content/browser/devtools/protocol/service_worker_handler.h"
 #include "content/browser/devtools/protocol/storage_handler.h"
 #include "content/browser/devtools/protocol/target_handler.h"
-#include "content/browser/devtools/protocol/tracing_handler.h"
 #include "content/browser/frame_host/navigation_handle_impl.h"
 #include "content/browser/frame_host/navigation_request.h"
 #include "content/browser/frame_host/render_frame_host_impl.h"
@@ -325,8 +324,6 @@ bool RenderFrameDevToolsAgentHost::Attac
       session->client()->MayReadLocalFiles()));
   session->AddHandler(std::make_unique<protocol::SecurityHandler>());
   if (!frame_tree_node_ || !frame_tree_node_->parent()) {
-    session->AddHandler(std::make_unique<protocol::TracingHandler>(
-        frame_tree_node_, GetIOContext()));
   }
 #if !defined(OS_ANDROID)
   session->AddHandler(std::make_unique<protocol::WebAuthnHandler>());
@@ -391,8 +388,6 @@ void RenderFrameDevToolsAgentHost::Ready
     NavigationHandle* navigation_handle) {
   NavigationHandleImpl* handle =
       static_cast<NavigationHandleImpl*>(navigation_handle);
-  for (auto* tracing : protocol::TracingHandler::ForAgentHost(this))
-    tracing->ReadyToCommitNavigation(handle);
 
   if (handle->frame_tree_node() != frame_tree_node_) {
     if (ShouldForceCreation() && handle->GetRenderFrameHost() &&
@@ -500,8 +495,6 @@ void RenderFrameDevToolsAgentHost::Rende
 
 void RenderFrameDevToolsAgentHost::FrameDeleted(RenderFrameHost* rfh) {
   RenderFrameHostImpl* host = static_cast<RenderFrameHostImpl*>(rfh);
-  for (auto* tracing : protocol::TracingHandler::ForAgentHost(this))
-    tracing->FrameDeleted(host);
   if (host->frame_tree_node() == frame_tree_node_) {
     DestroyOnRenderFrameGone();
     // |this| may be deleted at this point.
@@ -753,8 +746,6 @@ void RenderFrameDevToolsAgentHost::Synch
   if (!frame_trace_recorder_)
     return;
   bool did_initiate_recording = false;
-  for (auto* tracing : protocol::TracingHandler::ForAgentHost(this))
-    did_initiate_recording |= tracing->did_initiate_recording();
   if (did_initiate_recording) {
     frame_trace_recorder_->OnSynchronousSwapCompositorFrame(frame_host_,
                                                             frame_metadata);
--- a/content/browser/webui/content_web_ui_controller_factory.cc
+++ b/content/browser/webui/content_web_ui_controller_factory.cc
@@ -13,7 +13,6 @@
 #include "content/browser/net/network_errors_listing_ui.h"
 #include "content/browser/process_internals/process_internals_ui.h"
 #include "content/browser/service_worker/service_worker_internals_ui.h"
-#include "content/browser/tracing/tracing_ui.h"
 #include "content/browser/webrtc/webrtc_internals_ui.h"
 #include "content/public/browser/storage_partition.h"
 #include "content/public/browser/web_contents.h"
@@ -78,7 +77,7 @@ ContentWebUIControllerFactory::CreateWeb
     return std::make_unique<ServiceWorkerInternalsUI>(web_ui);
   if (url.host_piece() == kChromeUINetworkErrorsListingHost)
     return std::make_unique<NetworkErrorsListingUI>(web_ui);
-#if !defined(OS_ANDROID)
+#if 0
   if (url.host_piece() == kChromeUITracingHost)
     return std::make_unique<TracingUI>(web_ui);
 #endif
--- a/chrome/browser/chrome_browser_main.cc
+++ b/chrome/browser/chrome_browser_main.cc
@@ -98,9 +98,6 @@
 #include "chrome/browser/shell_integration.h"
 #include "chrome/browser/site_isolation/site_isolation_policy.h"
 #include "chrome/browser/startup_data.h"
-#include "chrome/browser/tracing/background_tracing_field_trial.h"
-#include "chrome/browser/tracing/navigation_tracing.h"
-#include "chrome/browser/tracing/trace_event_system_stats_monitor.h"
 #include "chrome/browser/translate/translate_service.h"
 #include "chrome/browser/ui/javascript_dialogs/chrome_javascript_native_app_modal_dialog_factory.h"
 #include "chrome/browser/ui/profile_error_dialog.h"
@@ -874,11 +871,6 @@ void ChromeBrowserMainParts::PostMainMes
 
   system_monitor_ = performance_monitor::SystemMonitor::Create();
 
-  // TODO(sebmarchand): Allow this to be created earlier if startup tracing is
-  // enabled.
-  trace_event_system_stats_monitor_ =
-      std::make_unique<tracing::TraceEventSystemStatsMonitor>();
-
   // device_event_log must be initialized after the message loop. Calls to
   // {DEVICE}_LOG prior to here will only be logged with VLOG. Some
   // platforms (e.g. chromeos) may have already initialized this.
@@ -1114,12 +1106,6 @@ int ChromeBrowserMainParts::PreCreateThr
 #endif  // defined(OS_WIN)
 #endif  // BUILDFLAG(ENABLE_VR)
 
-  // Enable Navigation Tracing only if a trace upload url is specified.
-  if (parsed_command_line_.HasSwitch(switches::kEnableNavigationTracing) &&
-      parsed_command_line_.HasSwitch(switches::kTraceUploadURL)) {
-    tracing::SetupNavigationTracing();
-  }
-
 #if defined(OS_WIN) || defined(OS_MACOSX) || \
     (defined(OS_LINUX) && !defined(OS_CHROMEOS))
   metrics::DesktopSessionDurationTracker::Initialize();
@@ -1185,8 +1171,6 @@ void ChromeBrowserMainParts::PostCreateT
       base::BindOnce(&tracing::TracingSamplerProfiler::CreateOnChildThread));
 #endif
 
-  tracing::SetupBackgroundTracingFieldTrial();
-
   for (size_t i = 0; i < chrome_extra_parts_.size(); ++i)
     chrome_extra_parts_[i]->PostCreateThreads();
 }
--- a/chrome/browser/chrome_content_browser_client.cc
+++ b/chrome/browser/chrome_content_browser_client.cc
@@ -145,7 +145,6 @@
 #include "chrome/browser/subresource_filter/chrome_subresource_filter_client.h"
 #include "chrome/browser/sync_file_system/local/sync_file_system_backend.h"
 #include "chrome/browser/tab_contents/tab_util.h"
-#include "chrome/browser/tracing/chrome_tracing_delegate.h"
 #include "chrome/browser/translate/translate_service.h"
 #include "chrome/browser/ui/blocked_content/blocked_window_params.h"
 #include "chrome/browser/ui/blocked_content/popup_blocker.h"
@@ -4307,7 +4306,7 @@ ChromeContentBrowserClient::GetDevToolsB
 }
 
 content::TracingDelegate* ChromeContentBrowserClient::GetTracingDelegate() {
-  return new ChromeTracingDelegate();
+  return nullptr;
 }
 
 bool ChromeContentBrowserClient::IsPluginAllowedToCallRequestOSFileHandle(
--- a/chrome/browser/prefs/browser_prefs.cc
+++ b/chrome/browser/prefs/browser_prefs.cc
@@ -65,7 +65,6 @@
 #include "chrome/browser/ssl/chrome_ssl_host_state_delegate.h"
 #include "chrome/browser/ssl/ssl_config_service_manager.h"
 #include "chrome/browser/task_manager/task_manager_interface.h"
-#include "chrome/browser/tracing/chrome_tracing_delegate.h"
 #include "chrome/browser/ui/blocked_content/safe_browsing_triggered_popup_blocker.h"
 #include "chrome/browser/ui/browser_ui_prefs.h"
 #include "chrome/browser/ui/hats/hats_service.h"
@@ -580,7 +579,6 @@ void RegisterLocalState(PrefRegistrySimp
   BrowserProcessImpl::RegisterPrefs(registry);
   ChromeContentBrowserClient::RegisterLocalStatePrefs(registry);
   ChromeMetricsServiceClient::RegisterPrefs(registry);
-  ChromeTracingDelegate::RegisterPrefs(registry);
   component_updater::RegisterPrefs(registry);
   ExternalProtocolHandler::RegisterPrefs(registry);
   flags_ui::PrefServiceFlagsStorage::RegisterPrefs(registry);
--- a/chrome/browser/lifetime/browser_shutdown.cc
+++ b/chrome/browser/lifetime/browser_shutdown.cc
@@ -35,9 +35,7 @@
 #include "components/metrics/metrics_service.h"
 #include "components/prefs/pref_registry_simple.h"
 #include "components/prefs/pref_service.h"
-#include "components/tracing/common/tracing_switches.h"
 #include "content/public/browser/render_process_host.h"
-#include "content/public/browser/tracing_controller.h"
 #include "printing/buildflags/buildflags.h"
 #include "rlz/buildflags/buildflags.h"
 
--- a/chrome/browser/sessions/tab_loader.cc
+++ b/chrome/browser/sessions/tab_loader.cc
@@ -18,7 +18,6 @@
 #include "chrome/browser/ui/browser.h"
 #include "chrome/browser/ui/browser_finder.h"
 #include "components/favicon/content/content_favicon_driver.h"
-#include "content/public/browser/background_tracing_manager.h"
 #include "content/public/browser/navigation_controller.h"
 #include "content/public/browser/web_contents.h"
 #include "content/public/common/content_features.h"
@@ -29,15 +28,6 @@ using resource_coordinator::TabLoadTrack
 namespace {
 
 void BackgroundTracingTrigger() {
-  static content::BackgroundTracingManager::TriggerHandle trigger_handle_ = -1;
-  if (trigger_handle_ == -1) {
-    trigger_handle_ =
-        content::BackgroundTracingManager::GetInstance()->RegisterTriggerType(
-            "session-restore-config");
-  }
-  content::BackgroundTracingManager::GetInstance()->TriggerNamedEvent(
-      trigger_handle_,
-      content::BackgroundTracingManager::StartedFinalizingCallback());
 }
 
 const base::TickClock* GetDefaultTickClock() {
--- a/components/feedback/tracing_manager.cc
+++ b/components/feedback/tracing_manager.cc
@@ -10,7 +10,6 @@
 #include "base/single_thread_task_runner.h"
 #include "base/threading/thread_task_runner_handle.h"
 #include "components/feedback/feedback_util.h"
-#include "content/public/browser/tracing_controller.h"
 
 namespace {
 
@@ -41,10 +40,6 @@ int TracingManager::RequestTrace() {
 
   current_trace_id_ = g_next_trace_id;
   ++g_next_trace_id;
-  content::TracingController::GetInstance()->StopTracing(
-      content::TracingController::CreateStringEndpoint(
-          base::Bind(&TracingManager::OnTraceDataCollected,
-                     weak_ptr_factory_.GetWeakPtr())));
   return current_trace_id_;
 }
 
@@ -87,9 +82,6 @@ void TracingManager::DiscardTraceData(in
 }
 
 void TracingManager::StartTracing() {
-  content::TracingController::GetInstance()->StartTracing(
-      base::trace_event::TraceConfig(),
-      content::TracingController::StartTracingDoneCallback());
 }
 
 void TracingManager::OnTraceDataCollected(
--- a/chrome/browser/profiling_host/profiling_process_host.cc
+++ b/chrome/browser/profiling_host/profiling_process_host.cc
@@ -22,7 +22,6 @@
 #include "base/values.h"
 #include "build/build_config.h"
 #include "chrome/browser/browser_process.h"
-#include "chrome/browser/tracing/crash_service_uploader.h"
 #include "components/heap_profiling/supervisor.h"
 #include "components/services/heap_profiling/public/cpp/controller.h"
 #include "components/services/heap_profiling/public/cpp/settings.h"
@@ -48,20 +47,6 @@ const char kConfigCategoryKey[] = "categ
 const char kConfigCategoryMemlog[] = "MEMLOG";
 const char kOOPHeapProfilingUploadUrl[] = "upload_url";
 
-void OnTraceUploadComplete(TraceCrashServiceUploader* uploader,
-                           bool success,
-                           const std::string& feedback) {
-  UMA_HISTOGRAM_BOOLEAN("HeapProfiling.UploadTrace.Success", success);
-
-  if (!success) {
-    LOG(ERROR) << "Cannot upload trace file: " << feedback;
-    return;
-  }
-
-  // The reports is successfully sent. Reports the crash-id to ease debugging.
-  LOG(WARNING) << "slow-reports sent: '" << feedback << '"';
-}
-
 void UploadTraceToCrashServer(std::string upload_url,
                               std::string file_contents,
                               std::string trigger_name,
@@ -90,16 +75,6 @@ void UploadTraceToCrashServer(std::strin
       std::make_unique<base::DictionaryValue>();
   metadata->SetKey("config", std::move(configs));
   metadata->SetKey(kConfigScenarioName, base::Value("MEMLOG"));
-
-  TraceCrashServiceUploader* uploader = new TraceCrashServiceUploader(
-      g_browser_process->shared_url_loader_factory());
-  if (!upload_url.empty())
-    uploader->SetUploadURL(upload_url);
-
-  uploader->DoUpload(file_contents, content::TraceUploader::COMPRESSED_UPLOAD,
-                     std::move(metadata),
-                     content::TraceUploader::UploadProgressCallback(),
-                     base::Bind(&OnTraceUploadComplete, base::Owned(uploader)));
 }
 
 }  // namespace
--- a/components/heap_profiling/test_driver.cc
+++ b/components/heap_profiling/test_driver.cc
@@ -27,7 +27,6 @@
 #include "content/public/browser/browser_thread.h"
 #include "content/public/browser/render_process_host.h"
 #include "content/public/browser/system_connector.h"
-#include "content/public/browser/tracing_controller.h"
 
 namespace heap_profiling {
 
--- a/components/heap_profiling/supervisor.cc
+++ b/components/heap_profiling/supervisor.cc
@@ -14,7 +14,6 @@
 #include "components/services/heap_profiling/public/cpp/settings.h"
 #include "content/public/browser/browser_task_traits.h"
 #include "content/public/browser/browser_thread.h"
-#include "content/public/browser/tracing_controller.h"
 #include "services/resource_coordinator/public/cpp/memory_instrumentation/memory_instrumentation.h"
 #include "services/resource_coordinator/public/mojom/memory_instrumentation/memory_instrumentation.mojom.h"
 #include "services/resource_coordinator/public/mojom/service_constants.mojom.h"
@@ -114,6 +113,9 @@ void Supervisor::RequestTraceWithHeapDum
   DCHECK(content::BrowserThread::CurrentlyOn(content::BrowserThread::UI));
   DCHECK(HasStarted());
 
+  return;
+
+  /*
   if (content::TracingController::GetInstance()->IsTracing()) {
     DLOG(ERROR) << "Requesting heap dump when tracing has already started.";
     base::ThreadTaskRunnerHandle::Get()->PostTask(
@@ -163,6 +165,7 @@ void Supervisor::RequestTraceWithHeapDum
       GetBackgroundTracingConfig(anonymize),
       base::AdaptCallbackForRepeating(std::move(trigger_memory_dump_callback)));
   DCHECK(result);
+  */
 }
 
 void Supervisor::StartServiceOnIOThread(
--- a/chrome/browser/ui/tab_helpers.cc
+++ b/chrome/browser/ui/tab_helpers.cc
@@ -59,7 +59,6 @@
 #include "chrome/browser/sync/sessions/sync_sessions_router_tab_helper.h"
 #include "chrome/browser/sync/sessions/sync_sessions_web_contents_router_factory.h"
 #include "chrome/browser/tab_contents/navigation_metrics_recorder.h"
-#include "chrome/browser/tracing/navigation_tracing.h"
 #include "chrome/browser/translate/chrome_translate_client.h"
 #include "chrome/browser/ui/autofill/chrome_autofill_client.h"
 #include "chrome/browser/ui/blocked_content/popup_blocker_tab_helper.h"
@@ -92,7 +91,6 @@
 #include "components/offline_pages/buildflags/buildflags.h"
 #include "components/password_manager/core/browser/password_manager.h"
 #include "components/subresource_filter/core/browser/subresource_filter_features.h"
-#include "components/tracing/common/tracing_switches.h"
 #include "components/ukm/content/source_url_recorder.h"
 #include "content/public/browser/web_contents.h"
 #include "extensions/buildflags/buildflags.h"
@@ -357,9 +355,6 @@ void TabHelpers::AttachTabHelpers(WebCon
   if (predictors::LoadingPredictorFactory::GetForProfile(profile))
     predictors::LoadingPredictorTabHelper::CreateForWebContents(web_contents);
 
-  if (tracing::NavigationTracingObserver::IsEnabled())
-    tracing::NavigationTracingObserver::CreateForWebContents(web_contents);
-
   if (MediaEngagementService::IsEnabled())
     MediaEngagementService::CreateWebContentsObserver(web_contents);
 
--- a/content/browser/builtin_service_manifests.cc
+++ b/content/browser/builtin_service_manifests.cc
@@ -27,7 +27,6 @@
 #include "services/network/public/cpp/manifest.h"
 #include "services/resource_coordinator/public/cpp/manifest.h"
 #include "services/service_manager/public/cpp/manifest_builder.h"
-#include "services/tracing/manifest.h"
 
 namespace content {
 
@@ -74,7 +73,6 @@ const std::vector<service_manager::Manif
                   : service_manager::Manifest::ExecutionMode::
                         kOutOfProcessBuiltin),
           resource_coordinator::GetManifest(),
-          tracing::GetManifest(),
       }};
   return *manifests;
 }
--- a/services/service_manager/public/cpp/service_binding.cc
+++ b/services/service_manager/public/cpp/service_binding.cc
@@ -11,8 +11,6 @@
 #include "base/synchronization/lock.h"
 #include "mojo/public/cpp/bindings/remote.h"
 #include "services/service_manager/public/cpp/service.h"
-#include "services/tracing/public/cpp/traced_process.h"
-#include "services/tracing/public/mojom/traced_process.mojom.h"
 
 namespace service_manager {
 
@@ -169,12 +167,6 @@ void ServiceBinding::OnBindInterface(
     return;
   }
 
-  if (interface_name == tracing::mojom::TracedProcess::Name_) {
-    tracing::TracedProcess::OnTracedProcessRequest(
-        tracing::mojom::TracedProcessRequest(std::move(interface_pipe)));
-    return;
-  }
-
   service_->OnConnect(source_info, interface_name, std::move(interface_pipe));
 }
 
--- a/content/public/browser/BUILD.gn
+++ b/content/public/browser/BUILD.gn
@@ -409,7 +409,6 @@ jumbo_source_set("browser_sources") {
     "//services/network/public/mojom",
     "//services/resource_coordinator/public/cpp:resource_coordinator_cpp",
     "//services/service_manager/public/cpp",
-    "//services/tracing/public/cpp",
     "//services/video_capture/public/mojom",
     "//services/viz/public/mojom",
     "//third_party/webrtc/modules/desktop_capture",
--- a/services/service_manager/public/cpp/BUILD.gn
+++ b/services/service_manager/public/cpp/BUILD.gn
@@ -44,10 +44,6 @@ component("cpp") {
     "//url",
   ]
 
-  deps = [
-    "//services/tracing/public/cpp:traced_process",
-  ]
-
   defines = [
     "IS_SERVICE_MANAGER_CPP_IMPL",
 
--- a/content/browser/tracing/tracing_controller_impl.cc
+++ b/content/browser/tracing/tracing_controller_impl.cc
@@ -212,19 +212,9 @@ void TracingControllerImpl::AddAgents()
 }
 
 void TracingControllerImpl::ConnectToServiceIfNeeded() {
-  if (!coordinator_) {
-    GetSystemConnector()->BindInterface(tracing::mojom::kServiceName,
-                                        &coordinator_);
-    coordinator_.set_connection_error_handler(base::BindOnce(
-        [](TracingControllerImpl* controller) {
-          controller->coordinator_.reset();
-        },
-        base::Unretained(this)));
-  }
 }
 
 void TracingControllerImpl::DisconnectFromService() {
-  coordinator_ = nullptr;
 }
 
 // Can be called on any thread.
@@ -400,14 +390,6 @@ bool TracingControllerImpl::StartTracing
       std::make_unique<base::trace_event::TraceConfig>(trace_config);
 
   ConnectToServiceIfNeeded();
-  coordinator_->StartTracing(
-      trace_config.ToString(),
-      base::BindOnce(
-          [](StartTracingDoneCallback callback, bool success) {
-            if (!callback.is_null())
-              std::move(callback).Run();
-          },
-          std::move(callback)));
   // TODO(chiniforooshan): The actual success value should be sent by the
   // callback asynchronously.
   return true;
@@ -535,16 +517,8 @@ bool TracingControllerImpl::StopTracing(
       new mojo::DataPipeDrainer(this, std::move(data_pipe.consumer_handle)));
   if (agent_label.empty()) {
     // Stop and flush all agents.
-    coordinator_->StopAndFlush(
-        std::move(data_pipe.producer_handle),
-        base::BindRepeating(&TracingControllerImpl::OnMetadataAvailable,
-                            base::Unretained(this)));
   } else {
     // Stop all and flush a particular agent.
-    coordinator_->StopAndFlushAgent(
-        std::move(data_pipe.producer_handle), agent_label,
-        base::BindRepeating(&TracingControllerImpl::OnMetadataAvailable,
-                            base::Unretained(this)));
   }
   // TODO(chiniforooshan): Is the return value used anywhere?
   return true;
@@ -555,12 +529,6 @@ bool TracingControllerImpl::GetTraceBuff
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
 
   ConnectToServiceIfNeeded();
-  coordinator_->RequestBufferUsage(base::BindOnce(
-      [](GetTraceBufferUsageCallback callback, bool success, float percent_full,
-         uint32_t approximate_count) {
-        std::move(callback).Run(percent_full, approximate_count);
-      },
-      std::move(callback)));
   // TODO(chiniforooshan): The actual success value should be sent by the
   // callback asynchronously.
   return true;
--- a/content/child/BUILD.gn
+++ b/content/child/BUILD.gn
@@ -111,7 +111,6 @@ target(link_target_type, "child") {
     "//services/resource_coordinator/public/cpp/memory_instrumentation",
     "//services/service_manager/public/cpp",
     "//services/service_manager/public/mojom",
-    "//services/tracing/public/cpp",
     "//skia",
     "//storage/common",
     "//third_party/blink/public:blink_headers",
--- a/mojo/public/tools/bindings/chromium_bindings_configuration.gni
+++ b/mojo/public/tools/bindings/chromium_bindings_configuration.gni
@@ -40,7 +40,6 @@ _typemap_imports = [
   "//services/proxy_resolver/public/cpp/typemaps.gni",
   "//services/resource_coordinator/public/cpp/typemaps.gni",
   "//services/service_manager/public/cpp/typemaps.gni",
-  "//services/tracing/public/mojom/typemaps.gni",
   "//services/viz/privileged/cpp/typemaps.gni",
   "//services/viz/privileged/mojom/compositing/typemaps.gni",
   "//services/viz/public/cpp/compositing/typemaps.gni",
--- a/content/app/BUILD.gn
+++ b/content/app/BUILD.gn
@@ -30,7 +30,6 @@ template("implement_content_app") {
     "//crypto",
     "//services/service_manager/embedder",
     "//services/service_manager/public/mojom",
-    "//services/tracing/public/cpp",
     "//mojo/core/embedder",
     "//ppapi/buildflags",
     "//ui/base",
--- a/content/utility/BUILD.gn
+++ b/content/utility/BUILD.gn
@@ -50,8 +50,6 @@ jumbo_source_set("utility") {
     "//services/service_manager/public/mojom",
     "//services/shape_detection:lib",
     "//services/shape_detection/public/mojom",
-    "//services/tracing:lib",
-    "//services/tracing/public/mojom",
     "//services/video_capture:lib",
     "//services/video_capture/public/mojom:constants",
     "//third_party/blink/public:blink_headers",
--- a/chrome/browser/chrome_browser_main.h
+++ b/chrome/browser/chrome_browser_main.h
@@ -32,10 +32,6 @@ class StartupTimeBomb;
 class ShutdownWatcherHelper;
 class WebUsbDetector;
 
-namespace tracing {
-class TraceEventSystemStatsMonitor;
-}
-
 namespace performance_monitor {
 class SystemMonitor;
 }
@@ -162,11 +158,6 @@ class ChromeBrowserMainParts : public co
   // metrics they need.
   std::unique_ptr<performance_monitor::SystemMonitor> system_monitor_;
 
-  // The system stats monitor used by chrome://tracing. This doesn't do anything
-  // until tracing of the |system_stats| category is enabled.
-  std::unique_ptr<tracing::TraceEventSystemStatsMonitor>
-      trace_event_system_stats_monitor_;
-
   // Whether PerformPreMainMessageLoopStartup() is called on VariationsService.
   // Initialized to true if |MainFunctionParams::ui_task| is null (meaning not
   // running browser_tests), but may be forced to true for tests.
--- a/content/app/content_main_runner_impl.cc
+++ b/content/app/content_main_runner_impl.cc
@@ -76,7 +76,6 @@
 #include "services/service_manager/sandbox/sandbox_type.h"
 #include "services/service_manager/sandbox/switches.h"
 #include "services/service_manager/zygote/common/zygote_buildflags.h"
-#include "services/tracing/public/cpp/trace_startup.h"
 #include "third_party/blink/public/common/origin_trials/trial_token_validator.h"
 #include "ui/base/ui_base_paths.h"
 #include "ui/base/ui_base_switches.h"
@@ -501,7 +500,7 @@ int RunZygote(ContentMainDelegate* deleg
       command_line.GetSwitchValueASCII(switches::kProcessType);
   ContentClientInitializer::Set(process_type, delegate);
 
-#if !defined(OS_ANDROID)
+#if 0
   tracing::EnableStartupTracingIfNeeded();
 #endif  // !OS_ANDROID
 
@@ -695,7 +694,7 @@ int ContentMainRunnerImpl::Initialize(co
     SetContentClient(&empty_content_client_);
   ContentClientInitializer::Set(process_type, delegate_);
 
-#if !defined(OS_ANDROID)
+#if 0
     // Enable startup tracing asap to avoid early TRACE_EVENT calls being
     // ignored. For Android, startup tracing is enabled in an even earlier place
     // content/app/android/library_loader_hooks.cc.
--- a/content/utility/utility_service_factory.cc
+++ b/content/utility/utility_service_factory.cc
@@ -28,8 +28,6 @@
 #include "services/data_decoder/public/mojom/constants.mojom.h"
 #include "services/network/network_service.h"
 #include "services/service_manager/public/mojom/service.mojom.h"
-#include "services/tracing/public/cpp/tracing_features.h"
-#include "services/tracing/public/mojom/constants.mojom.h"
 #include "services/tracing/tracing_service.h"
 
 #if BUILDFLAG(ENABLE_LIBRARY_CDMS)
@@ -149,10 +147,6 @@ void UtilityServiceFactory::RunService(
     content::UtilityThread::Get()->EnsureBlinkInitialized();
     service =
         std::make_unique<data_decoder::DataDecoderService>(std::move(request));
-  } else if (service_name == tracing::mojom::kServiceName &&
-             !base::FeatureList::IsEnabled(
-                 features::kTracingServiceInProcess)) {
-    service = std::make_unique<tracing::TracingService>(std::move(request));
   } else if (service_name == mojom::kNetworkServiceName) {
     // Unlike other services supported by the utility process, the network
     // service runs on the IO thread and never self-terminates.
--- a/chrome/browser/metrics/chrome_metrics_service_client.cc
+++ b/chrome/browser/metrics/chrome_metrics_service_client.cc
@@ -53,7 +53,6 @@
 #include "chrome/browser/safe_browsing/certificate_reporting_metrics_provider.h"
 #include "chrome/browser/sync/device_info_sync_service_factory.h"
 #include "chrome/browser/sync/profile_sync_service_factory.h"
-#include "chrome/browser/tracing/background_tracing_metrics_provider.h"
 #include "chrome/browser/translate/translate_ranker_metrics_provider.h"
 #include "chrome/common/buildflags.h"
 #include "chrome/common/channel_info.h"
@@ -686,9 +685,6 @@ void ChromeMetricsServiceClient::Registe
           g_browser_process->component_updater()));
 
   metrics_service_->RegisterMetricsProvider(
-      std::make_unique<tracing::BackgroundTracingMetricsProvider>());
-
-  metrics_service_->RegisterMetricsProvider(
       std::make_unique<metrics::DemographicMetricsProvider>(
           std::make_unique<ProfileClientImpl>()));
 
--- a/content/browser/browser_main_runner_impl.cc
+++ b/content/browser/browser_main_runner_impl.cc
@@ -18,11 +18,8 @@
 #include "base/trace_event/heap_profiler_allocation_context_tracker.h"
 #include "base/trace_event/trace_event.h"
 #include "build/build_config.h"
-#include "components/tracing/common/trace_startup_config.h"
-#include "components/tracing/common/tracing_switches.h"
 #include "content/browser/browser_main_loop.h"
 #include "content/browser/notification_service_impl.h"
-#include "content/browser/tracing/tracing_controller_impl.h"
 #include "content/common/content_switches_internal.h"
 #include "content/public/common/content_switches.h"
 #include "content/public/common/main_function_params.h"
@@ -165,10 +162,6 @@ void BrowserMainRunnerImpl::Shutdown() {
 
   main_loop_->PreShutdown();
 
-  // Finalize the startup tracing session if it is still active.
-  if (TracingControllerImpl::GetInstance())
-    TracingControllerImpl::GetInstance()->FinalizeStartupTracingIfNeeded();
-
   {
     // The trace event has to stay between profiler creation and destruction.
     TRACE_EVENT0("shutdown", "BrowserMainRunner");
--- a/content/browser/service_manager/service_manager_context.cc
+++ b/content/browser/service_manager/service_manager_context.cc
@@ -607,16 +607,6 @@ ServiceManagerContext::ServiceManagerCon
 
   GetContentClient()->browser()->WillStartServiceManager();
 
-  if (base::FeatureList::IsEnabled(features::kTracingServiceInProcess)) {
-    RegisterInProcessService(tracing::mojom::kServiceName,
-                             base::CreateSequencedTaskRunner(
-                                 {base::ThreadPool(), base::MayBlock(),
-                                  base::TaskShutdownBehavior::SKIP_ON_SHUTDOWN,
-                                  base::WithBaseSyncPrimitives(),
-                                  base::TaskPriority::USER_BLOCKING}),
-                             base::BindRepeating(&CreateTracingService));
-  }
-
   if (IsInProcessNetworkService()) {
     scoped_refptr<base::SequencedTaskRunner> task_runner =
         service_manager_thread_task_runner_;
--- a/tools/binary_size/BUILD.gn
+++ b/tools/binary_size/BUILD.gn
@@ -19,6 +19,5 @@ python_library("sizes_py") {
   testonly = true
   pydeps_file = "sizes.pydeps"
   data_deps = [
-    "//third_party/catapult/tracing:convert_chart_json",
   ]
 }
--- a/content/test/BUILD.gn
+++ b/content/test/BUILD.gn
@@ -1160,7 +1160,6 @@ test("content_browsertests") {
     "//services/network:test_support",
     "//services/service_manager/public/cpp",
     "//services/test/echo/public/mojom",
-    "//services/tracing:privacy_check",
     "//services/video_capture/public/cpp",
     "//services/video_capture/public/cpp:mocks",
     "//services/video_capture/public/mojom:constants",
--- a/third_party/catapult/BUILD.gn
+++ b/third_party/catapult/BUILD.gn
@@ -102,9 +102,7 @@ group("telemetry_chrome_test_support") {
   ]
 
   data_deps += [
-    "third_party/gsutil",
     "third_party/typ",
-    "third_party/vinn",
   ]
 
   data += [
@@ -128,4 +126,4 @@ group("telemetry_chrome_test_support") {
     "tracing/tracing_examples/",
   ]
 
-}
\ No newline at end of file
+}
--- a/third_party/catapult/devil/BUILD.gn
+++ b/third_party/catapult/devil/BUILD.gn
@@ -6,7 +6,6 @@ group("devil") {
   testonly = true
   deps = []
   data_deps = [
-    "../third_party/gsutil",
   ]
   data = [
     "devil/",
